### PR TITLE
the checked pantry method, checks to see if the user has all the ingr…

### LINKED ISSUE #64: Iteration 4 - Pantry Data
Create classes and methods that can:

- [x] Determine whether a user’s pantry has enough ingredients to cook a given meal.
- [ ] Determine the amount of missing ingredients still needed to cook a given meal, based on what’s in the user’s pantry.

### PR DESCRIPTION
**Pull request by RosaTheDev**
…edients needed to cook the recipe

What (if any) features are you implementing?
- a method in the user class to check if the user has all the ingredients needed for cooking a recipe

What (if anything) did you refactor?
- ran eslint and also had it fix indentation issues

Were there any issues that arose?
-

Is there anything that you need from your teammate?
- It needs tests to along with it, I would love to go over how to break that code up into multiple tests

Any other comments, questions, or concerns?

### CODE REVIEW

File: src/classes/User.js [hunk 1]
```diff
@@ -63,5 +70,49 @@ class User {
...
+    console.log('user pantry', this.pantry)
+    
+    let findMatchingIngredient = recipeData.ingredients.map(recipeIngredient => {
+      let itIsInThePantry = this.pantry.find(pantryIngredient => {
+        return recipeIngredient.id === pantryIngredient.ingredient
+      })
+      
+      let hasAllIngredients = true;
+     
+      let determineAmountMissing = itIsInThePantry.amount - recipeIngredient.quantity.amount
```
Comment (devonwigle) [inline src/classes/User.js]: Needs  a test to check math

File: src/classes/User.js [hunk 2]
```diff
@@ -63,5 +70,49 @@ class User {
...
+
+      if (determineAmountMissing < 1)  {
+        determineAmountMissing = Math.abs(itIsInThePantry.amount - recipeIngredient.quantity.amount)
+        //inner text or html 
+        console.log('dont got it')
+      } else {
+        console.log('recipe', recipeIngredient)
+        console.log('user Pantry', itIsInThePantry)
+        console.log('You have all the ingredients you need');
+      }
```
Comment (devonwigle) [inline src/classes/User.js]: Another test for the response

File: src/classes/User.js [hunk 3]
```diff
@@ -63,5 +70,49 @@ class User {
...
+
+      // eslint-disable-next-line max-len
+      if (itIsInThePantry && (itIsInThePantry.amount  < recipeIngredient.quantity.amount) || !itIsInThePantry) {
+        hasAllIngredients = false;
+        // console.log('inside if stattement but false')
+        return hasAllIngredients
+      } else {
+        // console.log('inside if statement but true')
```
Comment (devonwigle) [inline src/classes/User.js]: Not sure this else is required since there is a return in the if statement

File: src/classes/User.js [hunk 4]
```diff
@@ -63,5 +70,49 @@ class User {
...
+      }
+    })
+
+    let hasfalse = findMatchingIngredient.includes(false)
+
+    if (hasfalse === true) {
+      console.log('OH NO! YOU dont have enough ingredients to cook this recipe');
+    } else {
+      // inner text
+      console.log('Happy cooking!');
```
Comment (devonwigle) [inline src/classes/User.js]: test for this response as well.

**Review by tommi-t-nguyen — DISMISSED** [review]
👍

File: src/classes/User.js (hunk 2 above)
Comment (RosaTheDev) [reply src/classes/User.js]: moved over to pantry class

**Pull request closed without merging**